clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -4104,7 +4104,8 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
       CmdArgs.push_back("-fsycl-explicit-simd");
 
     if (!Args.hasFlag(options::OPT_fsycl_early_optimizations,
-                      options::OPT_fno_sycl_early_optimizations, true))
+                      options::OPT_fno_sycl_early_optimizations, true) ||
+        Triple.getSubArch() == llvm::Triple::SPIRSubArch_fpga)
```

Comment:
This is disable early opts for FPGA all the time, regardless of passing `-fsycl-early-optimizations` on the command line. Is that OK?

Reply:
No. I realized that after opening the PR. I'll update the PR by changing default value in a second.

Reply:
Should be fixed by https://github.com/intel/llvm/pull/2331/commits/586d541bc3bb216df94f050ec2ef9147cbbfe16b.